feat: Migrate from @sentry/apm to @sentry/tracing (#2021)
getsentry/sentry-docs: src/includes/performance-monitoring/configuration/vue.mdx

**Vue.js**

To get started with performance monitoring with Vue.js, first install these packages:

```bash
# Using yarn
$ yarn add @sentry/browser @sentry/integrations @sentry/tracing

# Using npm
$ npm install @sentry/browser @sentry/integrations @sentry/tracing
```

<Alert level="warning" title="Note"><markdown>

We switched our package name from `@sentry/apm` to `@sentry/tracing`. If you were previously using `@sentry/apm`, please see [Switching to the tracing package](/product/performance/apm-to-tracing/)

</markdown></Alert>

The Vue Tracing Integration allows you to track rendering performance during an initial application load.

Sentry injects a handler inside Vue's `beforeCreate` mixin, providing access to a Vue component during its life cycle stages.
When Sentry encounters a component named `root`, which is a top-level Vue instance (as in `new Vue({})`), we use our AM BrowserTracing integration,

> and create a new span named `Vue Application Render`. Once the span has been created, it will wait until all of its child components render, and there aren't new rendering events triggered within the configured `timeout`, before marking the span as finished.

Reviewer: Spans don't have timeouts, do they? Is this actually a transaction?

Author: Ah this is based on `idleTransaction`. Maybe it is better just to remove this paragraph.

The described instrumentation functionality will give you very high-level information about the rendering performance of the Vue instance. However, the integration can also provide more fine-grained details about what actually happened during a specific activity.
To do that, you need to specify which components to track and what hooks to listen to (you can find a list of all available hooks [here](https://vuejs.org/v2/api/#Options-Lifecycle-Hooks)). You can also turn on tracking for all the components. However, it may be rather noisy if your app consists of hundreds of components. We encourage being more specific. If you don't provide hooks, Sentry will track a component's `mount` and `update` hooks.

> Note that we don't use `before` and `-ed` pairs for hooks, and you should provide a simple verb instead. For example, `update` is correct. `beforeUpdate` and `updated` are incorrect.

Reviewer: ```suggestion
Note that when specifying hooks we use the simple verb rather than `before` and `-ed` pairs. For example, `destroy` is correct. `beforeDestroy` and `destroyed` are incorrect.
```

NB that I'm only pattern matching - I don't know for sure if `destroy` has a `before` and an `-ed`. I just thought that if possible it might be nice to have one that isn't the default since they already have examples of us correctly formatting the default ones.

Author: @kamilogorek will know.

Reviewer 2: They all do, except `activate`. That's why we use one simple verbs instead. Here's the full mapping list - https://github.com/getsentry/sentry-javascript/blob/5ed313546dd687b637dfb20a3a28a7823278d21a/packages/integrations/src/vue.ts#L116-L123

To set up the Vue Tracing Integration, you will first need to configure the AM BrowserTracing integration itself. For details on how to do this, see the [JavaScript](/performance-monitoring/distributed-tracing/#javascript) section above.
Once you've configured the BrowserTracing integration, move on to configuring the Vue integration itself.
Sentry built the new tracing capabilities into the original Vue error handler integrations, so there is no need to add any new packages. You only need to provide an appropriate configuration.

> The most basic configuration for tracing your Vue app, which would track only the top-level component, looks like this:

Reviewer: ```suggestion
The most basic configuration for tracing your Vue app, which would track only the `mount` and `update` hooks for the top-level component, looks like this:
```

```js
import * as Sentry from "@sentry/browser";
import { Vue as VueIntegration } from "@sentry/integrations";

> import { Integrations } from "@sentry/tracing";

Reviewer: If we're importing the entire tracing package here so that the `Sentry.init()` can look the same as it does everywhere else, perhaps we should do the same for the Vue integration? It might not be a bad idea to have at least one example where we're gracefully handling the `Integrations` naming conflict.

(Alternatively, we could handle every instance of this the other way, and only ever import the `BrowserTracing` integration instead of the whole package, which would make the question of naming conflicts moot.)

Author: > only ever import the BrowserTracing integration instead of the whole package

We can make this change.

```bash
# Using yarn
$ yarn add @sentry/browser @sentry/integrations @sentry/tracing

# Using npm
$ npm install @sentry/browser @sentry/integrations @sentry/tracing
```

<Alert level="warning" title="Note"><markdown>

We switched our package name from `@sentry/apm` to `@sentry/tracing`. If you were previously using `@sentry/apm`, please see [Switching to @sentry/tracing](/product/performance/apm-to-tracing/).

</markdown></Alert>

The Vue Tracing Integration allows you to track rendering performance during an initial application load.

Sentry injects a handler inside Vue's `beforeCreate` mixin, providing access to a Vue component during its life cycle stages.

> When Sentry encounters a component named `root`, which is a top-level Vue instance (as in `new Vue({})`), we use our AM BrowserTracing integration,

Reviewer 3: > and there aren't new rendering events triggered within the configured `timeout`

Did this behavior change?

Author: I changed this because I felt that we didn't need to expose activities/`idleTransaction` concepts here.

If we really want to do this, we can do it in `BrowserTracing`.

In addition, with `@sentry/tracing`, the Vue integration now can work without `BrowserTracing` and `idleTransactions`, so referring to activities/`timeout` is not always correct.

  Vue,
  tracing: true,
  tracingOptions: {
    trackComponents: [
      "App",
      "RwvHeader",
      "RwvFooter",
      "RwvArticleList",
      "Pagination",
    ],

>     hooks: ["mount", "update", "destroy"],

Reviewer: IDK if a comment in the code here is the right place - perhaps it belongs in the explanatory text above - but it might be nice to mention that specifying the hooks yourself _replaces_ rather than adds to the default list, so `'mount'` and `'update'` need to be in there if they're still going to be tracked.

Author: This is a good point, but something we can address in a future PR.
